src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
 
+      case 'clear':
+        for (const key in currentState) {
+          delete currentState[key];
+        }
+        break;
+    }
```

Comment:
The `switch` statement is missing a `default` case. According to the task requirements, you need to include a `default` case to handle any action types that are not explicitly listed. This will ensure that the function behaves correctly even when unexpected action types are encountered.